### PR TITLE
ENH adding tests and docs for interp

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds some docs and tests for interp code.

### CODE REVIEW

**Comment by beckermr** [comment]
OK @esheldon This one is ready to go.

File: descwl_coadd/coadd.py [hunk 1]
```diff
@@ -320,8 +326,39 @@ def _make_coadds(self):
...
+
+    Parameters
+    ----------
+    exps : list of afw_image.ExposureF
+        The list of images to coadd.
+    noise_exps : list of afw_image.ExposureF
+        The list of noise images to coadd.
+    psf_exps : list of afw_image.ExposureF
+        the list of PSF images to coadd.
+    coadd_wcs,
```
Comment (beckermr) [inline descwl_coadd/coadd.py:344]:
```suggestion
    coadd_wcs : DM stack sky WCS object
        The WCS for the final coadd.
```

File: descwl_coadd/coadd.py [hunk 2]
```diff
@@ -320,8 +326,39 @@ def _make_coadds(self):
...
+    coadd_dims : 2-tuple of ints
```
Comment (beckermr) [inline descwl_coadd/coadd.py:345]:
```suggestion
    coadd_dims : 2-tuple of ints
        The dimensions of the coadd in (x, y)
```

File: descwl_coadd/coadd.py [hunk 3]
```diff
@@ -320,8 +326,39 @@ def _make_coadds(self):
...
+    psf_dims : 2-tuple of ints
```
Comment (beckermr) [inline descwl_coadd/coadd.py:346]:
```suggestion
    psf_dims : 2-tuple of ints
        The dimensions of the psf coadd in (x, y).
```

File: descwl_coadd/coadd.py [hunk 4]
```diff
@@ -382,37 +419,43 @@ def _make_coadds(self):
...
+        noise_data = self._make_warps(
+            exps=self.noise_exps,
+            dims=self.coadd_dims,
+            wcs=self.coadd_wcs,
+            dopsf=True,
+        )
+
+        # we need the weights in the coadds to be the same
+        # so we replace them here
+        psf_data["weights"] = image_data["weights"]
```
Comment (esheldon) [inline descwl_coadd/coadd.py:448]: I think we need to make it more clear in the commit  messages that this bug fix was put in

File: descwl_coadd/coadd.py [hunk 5]
```diff
@@ -49,7 +50,9 @@ class MultiBandCoadds(object):
     coadd_wcs: galsim wcs
         wcs for final cuadd
     coadd_dims: (nx, ny)
-        Currently doing x first rather than row, col
+        Dimensions of the main coadd. Currently doing x first rather than row, col
+    psf_dims: (nx, ny)
+        Dimensions of the PSF coadd. Currently doing x first rather than row, col
```
Comment (esheldon) [inline descwl_coadd/coadd.py:55]: It is used as ny, nx

what made you think it was nx, ny?

File: descwl_coadd/coadd.py [hunk 6]
```diff
@@ -64,7 +67,9 @@ class MultiBandCoadds(object):
     interp_bright: bool
         If True, mark BRIGHT as SAT and do interpolation
     replace_bright: bool
-        If True, replace BRIGHT with nois.
+        If True, replace BRIGHT with noise.
+    max_mask_frac : float
```
Comment (esheldon) [inline descwl_coadd/coadd.py:71]: this is not even used yet, not sure why I have it there

File: descwl_coadd/coadd.py (hunk 5 above)
Comment (beckermr) [reply descwl_coadd/coadd.py:55]: you wrote this "Currently doing x first rather than row, col" above.

File: descwl_coadd/coadd.py (hunk 6 above)
Comment (beckermr) [reply descwl_coadd/coadd.py:71]: kk i'll remove

File: descwl_coadd/coadd.py (hunk 4 above)
Comment (beckermr) [reply descwl_coadd/coadd.py:448]: I'll squash the history on merge and make a nice message

Comment (beckermr) [reply descwl_coadd/coadd.py:448]: We should probs start keeping a changelog too.

File: descwl_coadd/coadd.py (hunk 5 above)
Comment (beckermr) [reply descwl_coadd/coadd.py]: fixed

**Comment by beckermr** [comment]
ok @esheldon back to you

**Pull request merged**